File: c/easy/minimum_time_visiting_all_points.c
#include <stdlib.h>

int minTimeToVisitAllPoints(int points[][2], int n) {
    int total_time = 0;

    for (int i = 0; i < n - 1; i++) {
        int x1 = points[i][0];
        int y1 = points[i][1];
        int x2 = points[i + 1][0];
        int y2 = points[i + 1][1];

        int dx = abs(x2 - x1);
        int dy = abs(y2 - y1);

        total_time += (dx > dy) ? dx : dy;
    }

    return total_time;
}

File: rust/easy/minimum_time_visiting_all_points.rs
impl Solution {
    pub fn min_time_to_visit_all_points(points: Vec<Vec<i32>>) -> i32 {
        let mut total_time = 0;

        for i in 0..points.len() - 1 {
            let dx = (points[i + 1][0] - points[i][0]).abs();
            let dy = (points[i + 1][1] - points[i][1]).abs();
            total_time += dx.max(dy);
        }

        total_time
    }
}
